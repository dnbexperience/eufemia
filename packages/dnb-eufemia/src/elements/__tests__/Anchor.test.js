/**
 * Element Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  toJson,
  axeComponent,
} from '../../core/jest/jestSetup'
import Component from '../Anchor'

const props = fakeProps(require.resolve('../Anchor'), {
  optional: true,
})
props.inner_ref = null
props.element = 'a'

describe('Anchor element', () => {
  it('have to match default Anchor element snapshot', () => {
    const Comp = mount(<Component {...props}>text</Component>)
    expect(toJson(Comp)).toMatchSnapshot()
  })
  it('has dnb-a class', () => {
    const Comp = mount(<Component>text</Component>)
    expect(Comp.find('.dnb-a').exists()).toBe(true)
  })
  it('has href', () => {
    const Comp = mount(<Component href="/url">text</Component>)
    expect(Comp.find('[href]').exists()).toBe(true)
  })
  it('has no-icon class when element was given', () => {
    const Comp = mount(
      <Component href="/url" target="_blank">
        <span>text</span>
      </Component>
    )
    expect(Comp.find('.dnb-anchor--no-icon').exists()).toBe(true)
  })
  it('has aria-describedby when target is blank', async () => {
    const Comp = mount(
      <Component href="/url" target="_blank" lang="en-GB">
        text
      </Component>
    )

    const id = Comp.find('a').instance().getAttribute('aria-describedby')
    expect(document.body.querySelector('#' + id).textContent).toBe(
      'Opens a new Window'
    )

    const title = 'External site'
    Comp.setProps({
      title,
    })

    expect(Comp.find('a').instance().getAttribute('title')).toBe(title)
    expect(document.body.querySelector('#' + id)).toBe(null)
  })
  it('should validate with ARIA rules as a Anchor element', async () => {
    const Comp = mount(<Component {...props} />)
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })
})
