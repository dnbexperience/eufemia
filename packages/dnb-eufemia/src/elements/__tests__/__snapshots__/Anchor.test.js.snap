// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor element have to match default Anchor element snapshot 1`] = `
<ForwardRef
  className="className"
  element="a"
  href="href"
  inner_ref={null}
  omitClass={true}
  skeleton="skeleton"
  target="target"
  target_blank_title="target_blank_title"
  to="to"
  tooltip={[Function]}
>
  <AnchorInstance
    className="className"
    element="a"
    href="href"
    inner_ref={null}
    omitClass={true}
    skeleton="skeleton"
    target="target"
    target_blank_title="target_blank_title"
    to="to"
    tooltip={[Function]}
  >
    <ForwardRef
      class={null}
      className="className"
      css={null}
      href="href"
      inner_ref={
        Object {
          "current": <a
            class="className dnb-a"
            href="href"
            target="target"
            to="to"
          >
            text
          </a>,
        }
      }
      internalClass={null}
      is="a"
      skeleton="skeleton"
      skeleton_method="font"
      target="target"
      target_blank_title="target_blank_title"
      to="to"
    >
      <ElementInstance
        class={null}
        className="className"
        css={null}
        href="href"
        inner_ref={
          Object {
            "current": <a
              class="className dnb-a"
              href="href"
              target="target"
              to="to"
            >
              text
            </a>,
          }
        }
        internalClass={null}
        is="a"
        skeleton="skeleton"
        skeleton_method="font"
        target="target"
        target_blank_title="target_blank_title"
        to="to"
      >
        <a
          className="className dnb-a"
          href="href"
          target="target"
          to="to"
        >
          text
        </a>
      </ElementInstance>
    </ForwardRef>
  </AnchorInstance>
</ForwardRef>
`;
